src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs

```diff
@@ -128,16 +128,9 @@ internal void SetResourceAndInitializeBatch(Resource resource)
                 process.Tags[key] = label.ToJaegerTag();
             }
 
-            if (serviceName != null)
-            {
-                process.ServiceName = serviceNamespace != null
-                    ? serviceNamespace + "." + serviceName
-                    : serviceName;
-            }
-
             if (string.IsNullOrEmpty(process.ServiceName))
             {
-                process.ServiceName = DefaultServiceName;
+                process.ServiceName = (string) this.ParentProvider.GetDefaultResource().Attributes.FirstOrDefault().Value;
```

Comment:
I believe this part is redundant with the instantiation of process.ServiceName in the constructor. The constructor sets it to the same thing as in this part, and we only change process.ServiceName if we successfully locate this within the Resource's attributes. I think we can remove the whole if statement from 131 on.

Reply:
marked outdated but still relevant concern if you have input @CodeBlanch . thanks!

Reply:
It is a bit strange to see it twice. I would just remove the serviceName off the ctor of Process and move the logic into SetResourceAndInitializeBatch. If it finds a serviceName on tags use that, otherwise use the default.

Reply:
Thanks for the advice, I've changed that and pushed it, will test locally. I forgot to mention earlier that this was before I realized that the if statement on line 131 does more than concatenate, it also is the location of us actually setting the process's servicename tag, so it is necessary to keep. This makes our perspective of the if (string.IsNullOrEmpty) clause on line 136 different, because now this does have a separate functionality than the one in the ctor.

Reply:
Process must be instantiated in the Ctor, which requires a servicename in all three of the Ctors for Process. So re-adding this back into the instantiation.

Reply:
OK if we leave it in ctor we should be able to remove from SetResourceAndInitializeBatch. How about this?
```csharp
            string serviceName = null;
            string serviceNamespace = null;
            foreach (var label in resource.Attributes)
            {
                string key = label.Key;

                if (label.Value is string strVal)
                {
                    switch (key)
                    {
                        case ResourceSemanticConventions.AttributeServiceName:
                            serviceName = strVal;
                            continue;
                        case ResourceSemanticConventions.AttributeServiceNamespace:
                            serviceNamespace = strVal;
                            continue;
                    }
                }

                if (process.Tags == null)
                {
                    process.Tags = new Dictionary<string, JaegerTag>();
                }

                process.Tags[key] = label.ToJaegerTag();
            }

            if (serviceName != null)
            {
                serviceName = serviceNamespace != null
                    ? serviceNamespace + "." + serviceName
                    : serviceName;
            }

            if (!string.IsNullOrEmpty(serviceName))
            {
                process.ServiceName = serviceName;
            }
```

I made it so if we don't find a new value it just keeps what was set in ctor.

Reply:
that works great, thank you for the smart solution.
changed `serviceNamespace != null` to string.isNullOrEmpty(serviceNamespace) but otherwise integrated as is.